### PR TITLE
No more check parameters from argc / argv but from the parser. Fix #184

### LINKED ISSUE #184: -mpi still needed with --read-config
Even if `-mpi` flag is included in the configure file used with `--read-config`, it is still needed to be added in the command executing `CoreNeuron`.
This is due to the fact that `-mpi` flag is checked if exists in the arguments of the executable.
If it is not provided, there simulation fails with `Segmentation Faults`

### PR DESCRIPTION
**Pull request by alkino**
We have to decide what to do with other arguments just like `-p4` that is not valid with the new parser.

Does someone know what is `nrn_music`?

### CODE REVIEW

File: coreneuron/mpi/nrnmpi.cpp [hunk 1]
```diff
@@ -73,7 +74,7 @@ void nrnmpi_init(int nrnmpi_under_nrncontrol, int* pargc, char*** pargv) {
                 b = true;
                 break;
             }
-            if (strcmp("--mpi", (*pargv)[i]) == 0) {
+            if (corenrn_param.mpi_enable) {
```
Comment (alexsavulescu) [inline coreneuron/mpi/nrnmpi.cpp:77]: Can the enclosing loop be modified to smtg like: 

```cpp 
if (corenrn_param.mpi_enable || corenrn_param.p4)
{
 b = true;
}
```

**Comment by alexsavulescu** [comment]
> Does someone know what is `nrn_music`?

I've seen this in NEURON, however I couldn't find any reference for its use. I believe it's a sort of NET events relayer. @nrnhines @pramodk would you know more about it?

File: coreneuron/mpi/nrnmpi.cpp (hunk 1 above)
Comment (alkino) [reply coreneuron/mpi/nrnmpi.cpp:77]: The problem with **p4** is that it does not exists anymore (coreneuron side at least), `-p` (for `--pattern`) take a file path.

**Comment by alkino** [comment]
Maybe I need to add `|| arg == "--mpi"` because there is maybe someone / somewhere giving it args!

File: coreneuron/mpi/nrnmpi.cpp (hunk 1 above)
Comment (alexsavulescu) [reply coreneuron/mpi/nrnmpi.cpp:77]: At the very least `corenrn_param.mpi_enable` can be checked outside the loop

Comment (alkino) [reply coreneuron/mpi/nrnmpi.cpp:77]: ahahaha of course, I'm so disappointed by this piece of code, that I miss that xD

**Comment by pramodk** [comment]
> Does someone know what is nrn_music?

Currently it's not but I wouldn't remove ( it could come back).

> Maybe I need to add || arg == "--mpi" because there is maybe someone / somewhere giving it args!

@alkino : wanted to ask same thing because NEURON (and neurodamus) pass arguments via CLI.

**Comment by alkino** [comment]
Endly remove all of this, as we think that `main1.cpp` is the only user of `nrnmpi_init`.

**Comment by alkino** [comment]
Pr. @nrnhines can we have a really quick and fast review of this?

Regarding callers of `nrnmpi_init` (only us?), and `nrn_music` (useless?).

Regards

**Comment by pramodk** [comment]
CoreNEURON will be either called by NEURON as library (where MPI is already initialised) or CoreNEURON will be launched independently. Hence extra options we had (e.g. always initialise MPI) are not necessary in my opinion.

Thanks for cleaning that up!

**Review by pramodk — APPROVED** [review]

**Comment by alkino** [comment]
`The base branch restricts merging to authorized users.` I leave authorized users do.

File: coreneuron/mpi/nrnmpi.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@ THE POSSIBILITY OF SUCH DAMAGE.
 #include <sys/time.h>
 
 #include "coreneuron/nrnconf.h"
+#include "coreneuron/apps/corenrn_parameters.hpp"
```
Comment (iomaganaris) [inline coreneuron/mpi/nrnmpi.cpp:33]: I think that this is not needed anymore?

**Review by iomaganaris — COMMENTED** [review]
LGTM. Just a small nitpick

File: coreneuron/mpi/nrnmpi.cpp (hunk 2 above)
Comment (pramodk) [reply coreneuron/mpi/nrnmpi.cpp]: yes! thanks Ioannis!

**Review by pramodk — APPROVED** [review]

File: coreneuron/mpi/nrnmpi.cpp (hunk 2 above)
Comment (alkino) [reply coreneuron/mpi/nrnmpi.cpp]: of course, sorry

**Review by iomaganaris — APPROVED** [review]

**Pull request merged**